warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -150,69 +155,113 @@ public Value reduce(Key key, Iterator<Value> iter) {
+                    // don't want to release any uid that we've already seen (i.e., with a newer timestamp) that has
+                    // been quarantined. If we're not paying attention to timestamps, then releasing a UID always takes
+                    // precedence over quarantining. In either case, if the UID is already in the uids list, then we
+                    // don't need to track it as released.
+                    for (String uid : v.getRELEASEUIDList()) {
+                        if ((timestampsIgnored || !quarantinedIds.contains(uid)) && !uids.contains(uid))
+                            releasedUids.add(uid);
+                        if (timestampsIgnored)
+                            quarantinedIds.remove(uid);
+                    }
+                }
+                
+                // Remove any UIDs in the REMOVEDUID list.
+                for (String uid : v.getREMOVEDUIDList()) {
+                    // Don't remove the UID if it's in the UID list since that means a newer key
+                    // (larger timestamp value) added the UID and we don't want to undo that add.
+                    // If timestampsIgnored is set, then we are presuming lots of collisions on
+                    // timestamp and the order of incoming values is non-deterministic. In that case
+                    // we give precedence to a removal over an add and mark this UID as removed even
+                    // if it was in the UID list.
+                    //
+                    // If we're not propagating changes then don't bother adding UIDs to the
+                    // REMOVEDUID list. This happens if either we're doing a full major compaction
+                    // or we've seen the ignore flag (e.g., we exceeded the max UID count). In
+                    // either of those cases, the output protocol buffer won't include any removed
+                    // UIDs so there's no point to collecting them here.
+                    if (propogate && (timestampsIgnored || (!uids.contains(uid) && !releasedUids.contains(uid))))
+                        uidsToRemove.add(uid);
+                    if (timestampsIgnored)
+                        uids.remove(uid);
+                }
+                
+                /*
+                 * This is added for backwards compatibility. The removal list was added to ensure that removals are propagated across compactions. In the case
+                 * where compactions did not occur, and the indices are converted into the newer protobuf, we must use the UID list to maintain removals for
+                 * deltas less than 0
+                 */
+                if (delta < 0 && v.getREMOVEDUIDList().isEmpty()) {
+                    for (String uid : v.getUIDList()) {
+                        // add to uidsToRemove, and decrement count if the uid is in UIDS
+                        uidsToRemove.add(uid);
+                        uids.remove(uid);
+                    }
+                }
```

Comment:
This block was effectively moved from an earlier version. @matthpeterson, are you aware of any cases where this is actually needed? I'm wondering if this can go too?

Reply:
I'm thinking that this backwards-compatibility code has been in place for so long now that any system having compacted its indexes since then would no longer need it.

Reply:
Ok, I'll push another commit that removes it.